### ISSUE TITLE
Smartypants preference setting is ignored

### ISSUE THREAD

**Issue by C8-DO** [MacDownApp/macdown#912]
Quotes are **always** changed into typographer's quotes, no matter whether the Smartypants option is activated or not. I have to change the corresponding settings in the system preferences to get straight quotes, which is quite annoying.

And regardless of the settings, inside of `code `sections quotes should **never** be changed. All programming languages (I know) use straight quotes, so this is how code should be rendered. And strings won't be color coded with typographer's quotes, either.

I am using MacDown Version 0.7.2d81 (951) right now. Not sure if this problem occurs in the current stable version, too.

**Comment by franklinyu** [comment]
I think you are talking to the editor pane. That is not handled by the preferences. If you enable Smartypants in system preference, MacDown would have *already* received curly quote *as soon as* you type the quote character; I don't think there is any way for MacDown to tell whether you intend curly quote or straight quote, since MacDown only sees curly quote in the document.

That checkbox in MacDown preference only matters if you have straight quote *in source file* (`.md` file), that is, if macOS *doesn't* convert them automatically. This way straight quote is preserved in source file, and MacDown has the opportunity to choose to render it as curly quote or straight quote. (Yes, MacDown can render straight quote as curly quote, without changing the source file.)

Speaking of disabling Smartypants in code section, that's a good idea. Need more research about the approach though.

**Comment by C8-DO** [comment]
I don't agree with you here. Straight and curly quotes are two different characters, and you can specifically enter them. On my (German) keyboard, if I type `Shift-2`, I get straight quotes, if I type `Option-Shift-2`I get curly quotes. The system settings for typographer's quotes catches the `Shift-2` and "translates" it to `Option-Shift-2`.

I *think* you should be able to do something similar in the editor pane (but I don't use Xcode, so I might be wrong). If smartypants in turned on, "translate" `Shift-2`(ASCII 32) into the appropriate character for the user's system language. If it's turned off, don't change the character. Or the other way round, if you receive the character for curly quotes in the system language and smartypants is turned off, translate it to ASCII 32. That's how I would basically go about it.

But nevertheless, smartypants currently does **not** work how you described. In my system settings typographer's quotes are turned on, in MacDown they're turned off. But the quotes are rendered as curly quotes, regardless.

![smartypants](https://user-images.githubusercontent.com/34969250/34695880-d0e7eae8-f4cc-11e7-82a3-a82cdde537c6.png)

After some more tests it gets even stranger. Even though I have turned off typographer's quotes in the system settings, the result stays the same. Only if I specifically select straight quotes in the "for Double Quotes" popup menu in System Preferences, I get straight quotes (and this time in the editor pane **and** the preview).

**Comment by C8-DO** [comment]
So after some more test the bottom line is this: no matter what the system settings are, and no matter what the smartypants settings are, the preview will always show the same quotes that the editor shows.

**Comment by franklinyu** [comment]
First of all, you meant [smart quote substitution](http://osxdaily.com/2014/04/10/change-quote-style-mac-os-x/) when you talked about the system preference for Smartypants, did you?

Assuming that is true, then the checkbox is not intended to change the behavior in editor pane, but the preview pane:
![screen-1](https://user-images.githubusercontent.com/3153452/34704714-00b32264-f4c9-11e7-9ecc-672c65570263.png) ![screen-2](https://user-images.githubusercontent.com/3153452/34704715-00bdd236-f4c9-11e7-864c-4224e4ee1ee5.png)
Note that in both screenshots, the editor pane has straight quotes instead of curly quotes. So the MacDown preference doesn’t *override* system preference; they are just independent settings, in charge of different behaviors.

If you would like to disable smart quote substitution in MacDown editor, while keep it enabled system-wide, we may add another option for it. (In this case, it belongs to the *Editor* tab of preference window, not *Markdown* tab.)

In any case, if there is already curly quote in editor pane, it is expected to be previewed as-is. Smart quote always translate straight quote to curly one, never the other way around. It doesn’t need to be “smart” to translate multiple characters into one…

**Comment by uranusjr** [comment]
(Not entirely related to this issue, just some rumbling.)

Maybe it’s for the best if we just remove the Smartypants extension. It is hacky to start with, works “meh” at best, and sometimes even causes troubles. macOS handles UTF-8 fine, and we’re always saving to UTF-8 anyway. People should just learn how to type smart quotes.

**Comment by franklinyu** [comment]
Then I suggest split this issue into three:

 1. Disable Smartypants in math and code section
 2. Application-wise smart quote substitution setting
 3. [Discussion] Remove Smartypants support

where the 3rd issue may close 1st.
